flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionJobVertex.java

```diff
@@ -269,19 +256,15 @@ public ExecutionJobVertex(
         return jobVertex.getOperatorIDs();
     }
 
-    public void setMaxParallelism(int maxParallelismDerived) {
+    public void setMaxParallelism(int maxParallelism) {
 
         Preconditions.checkState(
-                !maxParallelismConfigured,
+                isMaxParallelismAutoConfigured(),
```

Comment:
This is not sufficient based on the JobVertex API, because it is possible for the maxParallelism to both be configured manually and being auto-configured. There are no assertions to ensure only one approach was used.

Reply:
It also doesn't really make sense for the maxParallelism to only be settable if it was previously autoConfigured by the system.

Reply:
> It also doesn't really make sense for the maxParallelism to only be settable if it was previously autoConfigured by the system.

👍🏼 I was able to remove the concept of "auto-configured" into SchedulerBase, the only place it makes sense.